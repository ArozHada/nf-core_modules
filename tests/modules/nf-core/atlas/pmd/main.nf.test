nextflow_process {

    name "Test Process ATLAS_PMD"
    script "modules/nf-core/atlas/pmd/main.nf"
    process "ATLAS_PMD"
    config "tests/modules/nf-core/atlas/pmd/nextflow.config"
    tag "atlas_pmd"

    test("Human Illumina PE Sorted BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                            []
                           ]
                input[1] = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                input[2] = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
