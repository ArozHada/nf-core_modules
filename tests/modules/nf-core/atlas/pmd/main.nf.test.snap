{
    "Human Illumina PE Sorted BAM": {
        "content": [
            [
                "test.paired_end.sorted_PMD_Table.txt:md5,b28a9f4784f30a711046708eb7d94642",
                "test.paired_end.sorted_PMD_Table_counts.txt:md5,1aabfb83cfff6e0413f52d016c943e3e",
                "test.paired_end.sorted_PMD_input_Empiric.txt:md5,8ff876f50ce05dc9aac0646a31682df0",
                "test.paired_end.sorted_PMD_input_Exponential.txt:md5,2421450ef643316f4eacf5cb811e4df7",
                "versions.yml:md5,abec49256c52cb677fa0dad3473ed5a6"
            ]
        ],
        "timestamp": "2023-04-07T21:01:50+0000"
    }
}